src/Compilers/CSharp/Portable/Binder/Binder_Attributes.cs

```diff
@@ -174,6 +179,74 @@ private BoundAttribute BindAttributeCore(AttributeSyntax node, NamedTypeSymbol a
                 boundNamedArguments, resultKind, attributeType, hasErrors: resultKind != LookupResultKind.Viable);
         }
 
+        private void CheckArgumentNullability(MethodSymbol attributeConstructor, AnalyzedAttributeArguments arguments, DiagnosticBag diagnostics)
+        {
+            // Check constructor arguments
+            for (int i = 0; i < attributeConstructor.ParameterCount && i < arguments.ConstructorArguments.Arguments.Count; i++)
+            {
+                var parameter = attributeConstructor.Parameters[i];
+                var argument = arguments.ConstructorArguments.Argument(i);
+                reportPossibleNullabilityMismatch(parameter.Type, argument);
+            }
+
+            // Check assignments
+            for (int i = 0; i < arguments.NamedArguments.Length; i++)
+            {
+                var assignment = arguments.NamedArguments[i] as BoundAssignmentOperator;
+                TypeSymbolWithAnnotations assignmentType;
+                switch (assignment.Left)
+                {
+                    case BoundPropertyAccess propertyAccess:
+                        assignmentType = propertyAccess.PropertySymbol.Type;
+                        break;
+                    case BoundFieldAccess fieldAccess:
+                        assignmentType = fieldAccess.FieldSymbol.Type;
+                        break;
+                    default:
+                        throw ExceptionUtilities.Unreachable;
+                }
+
+                reportPossibleNullabilityMismatch(assignmentType, assignment.Right);
+            }
+
+            void reportPossibleNullabilityMismatch(TypeSymbolWithAnnotations destination, BoundExpression sourceExpression)
+            {
+                if (IsNullableEnabled(sourceExpression.SyntaxTree, sourceExpression.Syntax.SpanStart))
```

Comment:
>IsNullableEnabled [](start = 20, length = 17)

Why do we check if nullable annotations enabled? It feels to me that we should check language version instead (whether the nullable feature is enabled).

Reply:
We only call this method if the lang version supports nullable annotations. Once we're in here we know it does. I can add an assert to make that clearer.